### ISSUE TITLE
Pylance Unnecessary isinstance on Protocol

### ISSUE THREAD

**Issue by PASUNX** [microsoft/pylance-release#1496]
<h3> Environment data </h3>

-   Language Server version: 2021.6.3
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: strict

<h3> Code Snippet / Additional information </h3>

```python
from typing import Protocol, ClassVar, runtime_checkable

@runtime_checkable
class Foo(Protocol):

    text: ClassVar[str] = "FOO"

class Bar:

    text: ClassVar[str] = "BAR"

class Baz:

    text: ClassVar[str] = "BAZ"

def SayBarBaz(*objs: Foo):
    for obj in objs:
        if isinstance(obj, (Bar, Baz)): # Unnecessary isinstance call; "Foo" is never an instance of "Bar | Baz" Pylance(reportUnnecessaryIsInstance)
            print(obj.text)

SayBarBaz(Bar(), Baz())
```

```console
$ mypy protocol.py 
Success: no issues found in 1 source file
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202170-6-july-2021
